Add saunafs version name on code

rolysr opened the pull request:
This PR is the first of a series of changes related to modifying current uses of `saunafsVersion(M, m, p)` on the codebase by the corresponding feature/change/improvement introduced a given version `vM.m.p`. The intention is to make clearer the logic on some parts of the code, depending on the specific version of SaunaFS being used.

rolysr committed:
chore: Add alias for `saunafsVersion(4,8,0)`

This commit replaces occurrences of the hardcoded
`saunafsVersion(4,8,0)` in the code with a descriptive
constant, specifically for the introduction of the
`.saunafs_path_by_inode` hidden file.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

lgsilva3087 commented on src/common/saunafs_version.h:
```diff
@@ -47,3 +47,6 @@ constexpr uint32_t kFirstECVersion = saunafsVersion(3, 9, 5);
 constexpr uint32_t kACL11Version = saunafsVersion(3, 11, 0);
 constexpr uint32_t kRichACLVersion = saunafsVersion(3, 12, 0);
 constexpr uint32_t kEC2Version = saunafsVersion(3, 13, 0);
+constexpr uint32_t kPathByInodeHiddenFileVersion = saunafsVersion(4, 8, 0);
+constexpr uint32_t kUseQuotaInVolumeSizeVersion = saunafsVersion(4, 9, 0);
+constexpr uint32_t kMountInfoOnMonitoringVersion = saunafsVersion(5, 0, 0);
```
Consider:
```cpp
constexpr uint32_t kFirstVersionWithPathByInodeHiddenFile = saunafsVersion(4, 8, 0);
constexpr uint32_t kFirstVersionWithUseQuotaInVolumeSize = saunafsVersion(4, 9, 0);
constexpr uint32_t kFirstVersionWithMountInfoInMonitoring = saunafsVersion(5, 0, 0);
```

Then, the comparisons looks like:
```cpp
if (masterversion >= kFirstVersionWithPathByInodeHiddenFile) {...}
```

rolysr replied:
Done.

lgsilva3087 reviewed (approved):
LGTM. Please see my suggestion.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -2826,7 +2826,7 @@ uint8_t fs_setacl(inode_t inode, uint32_t uid, uint32_t gid, AclType type,
 uint8_t fs_fullpath(inode_t inode, uint32_t uid, uint32_t gid, std::string &fullPath) {
 	threc *rec = fs_get_my_threc();
-	if (masterversion < saunafsVersion(4, 8, 0)) {
+	if (masterversion < kPathByInodeHiddenFileVersion) {
```
Consider applying this suggestion from Gemini.

rolysr replied:
Done.

ralcolea commented on src/mount/mastercomm.cc:
```diff
@@ -3163,7 +3163,7 @@ uint8_t fs_makesnapshot(inode_t src_inode, inode_t dst_inode, const std::string
                           std::vector<QuotaEntry> &quotaEntries) {
 	threc *rec = fs_get_my_threc();
-	if (masterversion < saunafsVersion(4, 9, 0)) {
+	if (masterversion < kUseQuotaInVolumeSizeVersion) {
```
Consider applying this suggestion from Gemini.

rolysr replied:
Done.

ralcolea reviewed (approved):
LGTM. Please, consider applying the suggestions from @lgsilva3087 and Gemini.

rolysr committed:
chore: Add alias for `saunafsVersion(4,9,0)`

This commit replaces occurrences of the hardcoded
`saunafsVersion(4,9,0)` in the code with a descriptive
constant, specifically for the introduction of the
`usequotainvolumesize` option on the client.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

rolysr committed:
chore: Add alias for `saunafsVersion(5,0,0)`

This commit replaces occurrences of the hardcoded
`saunafsVersion(5,0,0)` in the code with a descriptive
constant, specifically for the introduction of displaying
the content of `.saunafs_mount_info` on the monitoring
service for each mountpoint. Also, this commit includes a
similar change on the CGI monitoring module regarding
naming this hardcoded version.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>
